pkgs/applications/science/math/mathematica/default.nix

```diff
@@ -34,139 +42,101 @@
 , xkeyboard_config
 , xorg
 , zlib
+# settings and optional packages
+, cudaSupport ? false
+, cudaPackages ? null
 , lang ? "en"
+, version ? null
 }:
 
-let
-  l10n = import ./l10ns.nix {
-    inherit lib requireFile lang;
-  };
-in stdenv.mkDerivation {
-  inherit (l10n) version name src;
+let versions = import ./versions.nix { inherit lib requireFile; };
+
+    matching-versions =
+      with lib; sort compareVersions (filter
+        (v: v.lang == lang
+            && (if version == null then true else isMatching v.version version))
+        versions);
+
+    found-version =
+      if matching-versions == []
+      then throw ("No registered Mathematica version found to match"
+                  + " version=${version} and language=${lang}")
+      else lib.head matching-versions;
+
+    drv-name = ./. + "/${lib.versions.major found-version.version}.nix";
+
+    real-drv = if lib.pathExists drv-name then drv-name else ./generic.nix;
 
-  nativeBuildInputs = [
+    isMatching = v1: v2:
+      let as = lib.versions.splitVersion v1;
+          bs = lib.versions.splitVersion v2;
+          n  = lib.min (lib.length as) (lib.length bs);
+      in lib.take n as == lib.take n bs;
+
+    compareVersions = v1: v2:
+      let a = lib.versions.major v1.version;
+          b = lib.versions.major v2.version;
+      in lib.toInt a > lib.toInt b;
+
+in
+
+import real-drv {
```

Comment:
Please use callPackage instead to no list **every** package here.

Reply:
Okay, I see. With `callPackage` I can resolve the ellipses too. Making a change here.